modules/agentless-s3-bucket/main.tf

```diff
@@ -0,0 +1,73 @@
+locals {
+  dd_tags = {
+    Datadog                 = "true"
+    DatadogAgentlessScanner = "true"
+  }
+}
+
+data "aws_region" "current" {}
+
+resource "aws_s3_bucket" "bucket" {
+  bucket_prefix = "datadog-agentless-scanning-"
+  tags          = merge(var.tags, local.dd_tags)
+}
+
+resource "aws_s3_bucket_public_access_block" "bucket_access_block" {
+  bucket = aws_s3_bucket.bucket.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "bucket_lifecycle" {
+  bucket = aws_s3_bucket.bucket.id
+  rule {
+    status = "Enabled"
+    id     = "expire_all_files"
+    expiration {
+      days = 2
+    }
+  }
+}
+
+data "aws_iam_policy_document" "bucket_access_policy_document" {
+  # TODO: add statement to deny access to everyone except the scanner roles
```

Comment:
shouldn't be necessary (there's an implicit deny)

Reply:
I think I tested and I had access to it as a User of the account. I will double check this.

Reply:
If the principal making the request has an identity-based policy with an explicit allow, it will override the implicit deny of the resource-based policy.

I think it's pretty common to grant S3 read permissions widely to certain roles (e.g. we do it at Datadog for engineers). Given that these buckets are expected to contain very sensitive data, it makes sense to also have a very restrictive bucket policy.

Reply:
Definitely can make sense as an hardening practice, as long as we're not worried it could block further integrations at some point (e.g. if one day we decide that our "hosted" AWS integration should be able to scan RDS instances and access this bucket)